core/embed/rust/src/trezorhal/display.rs

```diff
@@ -74,51 +80,51 @@ pub fn text_baseline(font: i32) -> i16 {
     unsafe { ffi::font_baseline(font).try_into().unwrap_or(i16::MAX) }
 }
 
-pub fn bar(x: i16, y: i16, w: i16, h: i16, fgcolor: u16) {
-    unsafe { ffi::display_bar(x.into(), y.into(), w.into(), h.into(), fgcolor) }
-}
-
-pub fn bar_radius(x: i16, y: i16, w: i16, h: i16, fgcolor: u16, bgcolor: u16, radius: u8) {
+#[inline(always)]
+#[cfg(all(feature = "disp_i8080_16bit_dw", not(feature = "disp_i8080_8bit_dw")))]
+pub fn pixeldata(c: u16) {
     unsafe {
-        ffi::display_bar_radius(
-            x.into(),
-            y.into(),
-            w.into(),
-            h.into(),
-            fgcolor,
-            bgcolor,
-            radius,
-        )
+        ffi::DISPLAY_DATA_ADDRESS.write_volatile(c);
     }
 }
 
-pub fn bar_radius_buffer(x: i16, y: i16, w: i16, h: i16, radius: u8, buffer: &mut BufferText) {
+#[cfg(feature = "framebuffer")]
+pub fn get_fb_addr() -> u32 {
```

Comment:
Returning integer seems wrong. Let's use something like
```
struct FrameBuffer(*mut u32)
```

Is there a situation where we'd want to work with multiple framebuffers? If not perhaps we can keep the pointer private to this module?

Reply:
we will most likely want to use 2 framebuffers, but probably not simultaneously. 

Problem here is that need to use the address in the inlined `pixel` function, and there i don't want to call the ffi as it would decrease performance.

added the struct here: https://github.com/trezor/trezor-firmware/commit/c37f89d7113ecd773327d2840223e44e7edd829c